consumer/consumererror/permanent.go

```diff
@@ -14,27 +14,60 @@
 
 package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
 
-import "errors"
+import (
+	"errors"
+	"strconv"
+)
 
-// permanent is an error that will be always returned if its source
+// NO_DATA_COUNT indicates an unknown quantity of telemetry items.
+const NO_DATA_COUNT = -1
+
+// Permanent is an error that will always be returned if its source
 // receives the same inputs.
-type permanent struct {
-	err error
+type Permanent struct {
```

Comment:
Does this need to be public? Would it be possible to return the count of rejected as part of `IsPermanent` and keep this private? Not sure if the ergonomics make sense that way

Reply:
Most uses of `IsPermanent` take advantage of the `bool` return type and look like `if IsPermanent(err) { ... }`, so I think this would ultimately reduce the ergonomics of the function. I think having the type public with a method to get the rejected items count is likely the best approach, but we can investigate alternatives if it will present a problem.